app/components/arclight/search_result_title_component.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Arclight
+  # Render a document title for a search result
+  class SearchResultTitleComponent < Blacklight::DocumentTitleComponent
+    def initialize(compact:, **args)
+      @compact = compact
+      super(**args)
+    end
+
+    def compact?
+      @compact
+    end
+
+    # Content for the document actions area
+    def actions # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+      return [] unless @actions
+
+      if block_given?
+        @has_actions_slot = true
+        return super
+      end
+
+      (@has_actions_slot && get_slot(:actions)) ||
```

Comment:
Could you add a comment about what is happening here? I figured it out but a little hard to dive into.